any_parser/any_parser.py

```diff
@@ -3,254 +3,209 @@
 
 PUBLIC_SHARED_BASE_URL = "https://public-api.cambio-ai.com"
 TIMEOUT = 60
 
 
-class ProcessType(Enum):
-    EXTRACT_PII = "extract_pii"
-    EXTRACT_TABLES = "extract_tables"
-    EXTRACT_KEY_VALUE = "extract_key_value"
-    EXTRACT_RESUME_KEY_VALUE = "extract_resume_key_value"
-    PARSE = "parse"
-    PARSE_WITH_OCR = "parse_with_ocr"
-    PARSE_WITH_LAYOUT = "parse_with_layout"
+def handle_parsing(func):
+    """Decorator to handle common file processing logic."""
 
+    def wrapper(
+        self,
+        file_path=None,
+        file_content=None,
+        file_type=None,
+        *args,
+        **kwargs,
+    ):
+        # pylint: disable=too-many-arguments
+        # Validate inputs
+        is_valid, error_message = validate_parser_inputs(
+            file_path=file_path,
+            file_content=file_content,
+            file_type=file_type,
+        )
 
-class AnyParser:
-    """AnyParser RT: Real-time parser for any data format."""
+        if not is_valid:
+            return error_message, ""
+
+        # Encode the file content in base64 if file_path is provided
+        if file_content is None:
+            assert file_path is not None  # Type narrowing for mypy
+            try:
+                with open(file_path, "rb") as file:
+                    file_content = base64.b64encode(file.read()).decode("utf-8")
+                    file_type = Path(file_path).suffix.lower().lstrip(".")
+            except Exception as e:
+                return f"Error: {e}", ""
+
+        return func(
+            self,
+            file_path=file_path,
+            file_content=file_content,
+            file_type=file_type,
+            *args,
+            **kwargs,
+        )
 
-    def __init__(self, api_key: str, base_url: str = PUBLIC_SHARED_BASE_URL) -> None:
-        """Initialize the AnyParser RT object.
+    return wrapper
 
-        Args:
-            api_key (str): The API key for the AnyParser
-            url (str): The URL of the AnyParser RT API.
 
-        Returns:
-            None
-        """
-        self._sync_parse_url = f"{base_url}/parse"
-        self._sync_extract_pii = f"{base_url}/extract_pii"
-        self._sync_extract_tables = f"{base_url}/extract_tables"
-        self._sync_extract_key_value = f"{base_url}/extract_key_value"
-        self._sync_extract_resume_key_value = f"{base_url}/extract_resume_key_value"
-        self._sync_parse_with_ocr = f"{base_url}/parse_with_ocr"
-        self._async_upload_url = f"{base_url}/async/upload"
-        self._async_fetch_url = f"{base_url}/async/fetch"
-        self._api_key = api_key
-        self._headers = {
-            "Content-Type": "application/json",
-            "x-api-key": self._api_key,
-        }
-
-    def _get_sync_response(
-        self, url_endpoint: str, file_path: str, extract_args: Optional[Dict] = None
-    ) -> Tuple[Optional[requests.Response], str]:
-        """Extract full content from a file in real-time.
+def handle_async_parsing(func):
```

Comment:
similarly, `async_convert_b64_to_url`.

Also, qq that do we need to have two for sync and async and can be unify into a single decorator?